enterprise/coderd/prebuilds/metricscollector_test.go

```diff
@@ -307,6 +299,131 @@ func TestMetricsCollector(t *testing.T) {
 	}
 }
 
+// TestMetricsCollector_DuplicateTemplateNames validates a bug that we saw previously which caused duplicate metric series
+// registration when a template was deleted and a new one created with the same name (and preset name).
```

Comment:
AFAIK soft template deletion and subsequent naming conflicts for a new template are not the only way for a duplicate label set to occur. template name is one of three labels. I wonder if there are other opportunities for a labelset conflict like this. The only one I can think of right now would be if someone gives two presets in the same template version the same name. Is that a potential risk that we need to look into here?

Reply:
From `coderd/database/dump.sql`:

```
CREATE UNIQUE INDEX idx_unique_preset_name ON template_version_presets USING btree (name, template_version_id);
```

You can't have two presets with the same name in a template version.